### PR TITLE
fix(passwordbox): reveal button not working

### LINKED ISSUE #16026: [Gallery][Skia.GTK]PasswordBox- RevealButton functionality is not working.
### Current behavior

https://github.com/unoplatform/uno/assets/74563602/2e07c609-35a9-4db0-9d1c-bf1f28a18dc5

### Expected behavior

https://github.com/unoplatform/uno/assets/74563602/3c664bae-56e6-4998-abb5-003dda0dda51

### How to reproduce it (as minimally and precisely as possible)

1. Open the link https://unogallerycanary.z27.web.core.windows.net/
2. Click/tap Burger Menu on the left top right.
3. click/tap "PasswordBox" from the "Component" list.
4. Type Text for the password box.
5. Click on the RevealButton displaying on the right side on the text box.
6. Observe that when clicking revealbutton nothing is happening.

### Workaround

_No response_

### Works on UWP/WinUI

None

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #16026

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RuntimeTests/Tests/Uno_UI_Xaml_Core/Given_InputManager.cs [hunk 1]
```diff
@@ -250,7 +250,7 @@ public async Task When_ProtectedCursor_PointerCaptured()
 		var button1 = (Button)stackPanel.Children[0];
 		var button2 = (Button)stackPanel.Children[1];
 
-		SetProtectedCursor(button1, InputSystemCursor.Create(InputSystemCursorShape.IBeam));
+		button1.SetProtectedCursor(InputSystemCursor.Create(InputSystemCursorShape.IBeam));
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Tests/Uno_UI_Xaml_Core/Given_InputManager.cs:253]: I'd keep `SetProtectedCursor` as a method in RuntimeTests so we can call it when running WinUI SamplesApp.

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs [hunk 2]
```diff
@@ -231,6 +230,11 @@ protected override void OnApplyTemplate()
 				_deleteButton = new WeakReference<Button>(button);
 			}
 
+			if (_contentElement is { })
+			{
+				_contentElement.SetProtectedCursor(Microsoft/* UWP don't rename */.UI.Input.InputSystemCursor.Create(Microsoft/* UWP don't rename */.UI.Input.InputSystemCursorShape.IBeam));
+			}
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs:236]: This doesn't seem relevant to reveal button, and unclear what this change is trying to fix.

Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs:236]: It limits the IBeam cursor change to the "textbox body" and excludes the reveal/delete buttons.

File: src/Uno.UI.RuntimeTests/Tests/Uno_UI_Xaml_Core/Given_InputManager.cs (hunk 1 above)
Comment (ramezgerges) [reply src/Uno.UI.RuntimeTests/Tests/Uno_UI_Xaml_Core/Given_InputManager.cs:253]: It doesn't really matter. The entire class is already skia-only. If we change that in the future, we can add 2-liner back.

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs [hunk 3]
```diff
@@ -21,11 +21,12 @@ internal class TextBoxView
 
 		public TextBoxView(TextBox textBox)
 		{
+			_textBox = new WeakReference<TextBox>(textBox);
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs:24]: I failed to notice this one before, but we should be using the WeakReferencePool here. All DependencyObject instance have a pre-build weak ref, which avoid creating too many of those.

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs (hunk 3 above)
Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs]: @jeromelaban Does it need to be a WeakReference at all? There is no iOS circularity involved at all as this code is just Skia.

Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs]: Hmm, actually, TextBoxView's lifetime matches TextBox, so there is no reason to make the reference weak.

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs]: Yes, with the new pooling that's coming, it will be required

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs [hunk 4]
```diff
@@ -14,14 +12,13 @@ internal class TextBoxView
...
 
-		private readonly WeakReference<TextBox> _textBox;
 		private readonly bool _isPasswordBox;
 		private bool _isPasswordRevealed;
 		private readonly bool _isSkiaTextBox = !FeatureConfiguration.TextBox.UseOverlayOnSkia;
 
 		public TextBoxView(TextBox textBox)
 		{
-			_textBox = new WeakReference<TextBox>(textBox);
+			TextBox = textBox;
```
Comment (ajpinedam) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs:21]: Based on Jerome's comment on the other thread, I understand we should use the `WeakReferencePool` instead of the `WeakReference`

**Review by ajpinedam — APPROVED** [review]
